flypipe/node.py

```diff
@@ -83,6 +89,10 @@ def __init__(  # pylint: disable=too-many-arguments
         self.requested_columns = requested_columns
         self.node_graph = None
 
+        if cache is not None and not isinstance(cache, Cache):
+            raise TypeError("cache is not of type flypipe.cache.Cache")

```

Comment:
not really. the user has to create his own cache class and inherit from flypipe.cache.Cache. 
Without this assertion, it will pass objects not of type Cache and if he forgot to implement some abstract method, it will raise an exception.

It is more of a guide to implement his own cache

Reply:
Hmm ok, my comment would be similar to this https://github.com/flypipe/flypipe/pull/124#discussion_r1232903742, but I don't have a strong opinion on it.

Reply:
removed